### PR TITLE
adding new rules for keylogging, HTTP comms, network collection

### PR DESCRIPTION
**Pull request by mike-hunhoff**

### CODE REVIEW

File: host-interaction/network/connectivity/check-internet-connectivity-via-wininet.yml [hunk 1]
```diff
@@ -11,12 +11,14 @@ rule:
...
         - or:
-          - number: 0
-          - number: 1
+          - api: wininet.InternetGetConnectedState
+          - api: wininet.InternetCheckConnection
+        - optional:
+          - mnemonic: cmp
+          - or:
+            - number: 0
+            - number: 1
```
Comment (williballenthin) [inline host-interaction/network/connectivity/check-internet-connectivity-via-wininet.yml:23]: do you know what these constants represent?

File: nursery/get-tcp-endpoints.yml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# generated using capa explorer for IDA Pro
+rule:
+  meta:
+    name: get TCP endpoints
```
Comment (williballenthin) [inline nursery/get-tcp-endpoints.yml:4]: what do you mean by "endpoints"?

is this enumerating TCP connections? sessions? listeners?

File: nursery/get-tcp-endpoints.yml [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+    namespace: collection/network
+    author: michael.hunhoff@fireeye.com
+    scope: basic block
```
Comment (williballenthin) [inline nursery/get-tcp-endpoints.yml:7]:
```suggestion
    scope: basic block
    att&ck:
      - Discovery::Network Service Scanning [T1046]
```

Comment (williballenthin) [reply nursery/get-tcp-endpoints.yml:7]: not technically "scanning" though it fits the description

>  "Adversaries may attempt to get a listing of services running on remote hosts"

File: nursery/get-udp-endpoints.yml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# generated using capa explorer for IDA Pro
+rule:
+  meta:
+    name: get UDP endpoints
+    namespace: collection/network
+    author: michael.hunhoff@fireeye.com
+    scope: basic block
```
Comment (williballenthin) [inline nursery/get-udp-endpoints.yml:7]:
```suggestion
    scope: basic block
    att&ck:
      - Discovery::Network Service Scanning [T1046]
```

File: nursery/register-url.yml [hunk 5]
```diff
@@ -0,0 +1,11 @@
+# generated using capa explorer for IDA Pro
+rule:
+  meta:
+    name: register URL
```
Comment (williballenthin) [inline nursery/register-url.yml:4]: can we add a bit more context here? like "register URL with web server" maybe?

File: host-interaction/network/connectivity/check-internet-connectivity-via-wininet.yml (hunk 1 above)
Comment (mike-hunhoff) [reply host-interaction/network/connectivity/check-internet-connectivity-via-wininet.yml:23]: yes I believe these represent `TRUE` and `FALSE` - I'll get these added.

File: nursery/get-tcp-endpoints.yml (hunk 2 above)
Comment (mike-hunhoff) [reply nursery/get-tcp-endpoints.yml:4]: `GetExtendedTcpTable` can return connections, listeners, or both depending on which [TCP_TABLE_CLASS](https://docs.microsoft.com/en-us/windows/win32/api/iprtrmib/ne-iprtrmib-tcp_table_class) is specified. Maybe the name could be changed to `list TCP connections and listeners`?

File: nursery/get-tcp-endpoints.yml (hunk 3 above)
Comment (mike-hunhoff) [reply nursery/get-tcp-endpoints.yml:7]: hmm `GetExtendedTcpTable` only lists the local system's existing TCP connections and listeners which to me doesn't fit well under "listing of services running on remote hosts". Thoughts?

File: nursery/get-udp-endpoints.yml (hunk 4 above)
Comment (mike-hunhoff) [reply nursery/get-udp-endpoints.yml:7]: see `get TCP endpoints` comment

File: nursery/register-url.yml (hunk 5 above)
Comment (mike-hunhoff) [reply nursery/register-url.yml:4]: good catch - how about `register URL with HTTP server`?

File: nursery/get-tcp-endpoints.yml (hunk 3 above)
Comment (williballenthin) [reply nursery/get-tcp-endpoints.yml:7]: ok, thats fine by me

File: nursery/register-url.yml (hunk 5 above)
Comment (williballenthin) [reply nursery/register-url.yml:4]: sounds great

**Pull request merged**